### ISSUE TITLE
Documentation not being created for methods originating in inherited classes

### ISSUE THREAD

**Issue by drullo** [RicoSuter/Namotion.Reflection#8]
XML documentation placed on (abstract) base controller class does not appear in swagger doc.  Only XML documentation applied to the controller class that inherits the base class appears.

```[Route("api/[controller]")]
    [ApiController]
    public abstract class BaseController : ControllerBase
    {
        /// <summary>
        /// This documentation will not appear in swagger doc
        /// </summary>
        /// <param name="p">This will also not appear in swagger doc</param>
        /// <returns></returns>
        [HttpGet("{p}")]
        public ActionResult Test(int p)
        {
            return Ok();
        }
    }

    /// <summary>
    /// This controller now has a Patch method and an inherited Get method
    /// </summary>
    [Route("api/[controller]")]
    [ApiController]
    public class MyController : BaseController
    {
        /// <summary>
        /// This documentation WILL appear in swagger doc
        /// </summary>
        /// <param name="p">This parameter will be documented</param>
        /// <returns></returns>
        [HttpPatch("{p}")]
        public ActionResult Test2(int p)
        {
            return Ok();
        }
    }```

C#
.NET Core 2.2
Visual Studio 2017
NSwag 12.0.14

**Comment by RicoSuter** [comment]
Added tests, seems fine. Should be fixed with NSwag v13 and NJsonSchema v10.latest

**Comment by OrionDevelopment** [comment]
This isssue seems to persist, under NSwag.AspNetCore v. 13.0.6.
Has this been pushed downstream already?

**Comment by OrionDevelopment** [comment]
Figured out why it is not working in my case. My super class uses generics to define its methods in the base class, together with its documentation.

When you lookup the XML Documentation this fails for this exact reason.

**Comment by RicoSuter** [comment]
Can you provide a sample so that we can repro it here?

**Comment by OrionDevelopment** [comment]
The above example works, with a minor modification: add a generic argument to the class spec. Use this generic argument as a return type for the ActionResult, so ActionResult<TGeneric>. Observe the documentation dissapearing.

**Comment by RicoSuter** [comment]
If this is still sn issue, can someone create a pr with a unit test to repro this in this repository?

**Comment by DamienBraillard** [comment]
@RicoSuter HI Rico, I have been working on the issue for a few hours and I think I got it monstly solved (see pull request #29)

The problem arises when a controller inherits from a generic base and does not re-define it's methods like in the example below:
```csharp
        public class BaseGenericClass<T>
        {
            /// <summary>
            /// Single 
            /// </summary>
            public T Single(T input)
            {
                throw new NotImplementedException();
            }

            /// <summary>
            /// Multi 
            /// </summary>
            public ICollection<T> Multi(ICollection<T> input)
            {
                throw new NotImplementedException();
            }

            /// <summary>
            /// SingleAsync
            /// </summary>
            public Task<T> SingleAsync(T input)
            {
                throw new NotImplementedException();
            }

            /// <summary>
            /// MultiAsync
            /// </summary>
            public Task<ICollection<T>> MultiAsync(ICollection<T> input)
            {
                throw new NotImplementedException();
            }
        }

        public class InheritedGenericClass : BaseGenericClass<string>
        {
        }
```
The code is an extract of the UnitTest I wrote to check that it is fixed.

To make this work, I had to perform a quite extensive rewrite of the `Namotion.Reflection.XmlDocsExtensions.GetMemberElementName` method because a lot of special cases were not handled (ref parameters, array parameters, generic methods, etc...)

The fix does not apply partially for .Net Standard 1.0 not at all for Mono.Cecil.

For .Net Standard 1.0, the required reflection API to handle the generics are simply not present, I don't really see what I can do here. The .Net Standard 1.0 version benefits from other fixes though (ref parameters, array parameters, etc...).

Regarding Mono.Cecil, It should be doable but I don't know the library at all and have no clue how to port the code. For the moment, I have separated the implementation of GetMemberElementName for the Mono.Cecil implementation. It uses the old code so that nothing gets broken.

I understand that my proposed solution is not perfect but it is at least a step forward. It solves the problem for everyone using .Net Core 2.0+ and .Net framework 4.0+. If someone could take over fixing the Mono.Cecil part, this would be complete (in the exception of .Net Standard 1.0 for which I see no solution).
